### ISSUE TITLE
Unable to add a device on iot.tuya.com

### ISSUE THREAD

**Issue by BenGost** [codetheweb/tuyapi#334]
Hello,

When I add my plug connected via the tutorial, it does not appear in iot.tuya.com. I waited 48 hours and still the same.

MY outlet works well with the tuya mobile app. but cannot add it to the API.

However when I run the command: tuya-cli link --api-key <your api key> --api-secret <your api secret> --schema <your-schema / channel ID> --ssid <your WiFi name> --password <your WiFi password> --region eu

I do not have an error message, I have received the id, key and IP address of my home.

Sorry for my English, I am French.

Merci

--------------------------------
original text : 

Bonjour,

Quand j'ajoute ma prise connectée via le tutoriel, elle n'apparaît pas dans iot.tuya.com. J'ai attendu 48 heure et toujours pareil.

MA prise fonctionne bien avec l'application mobile tuya. mais impossible de l'ajouter à l'API.

Pourtant quand je lance la commande : tuya-cli link --api-key <your api key> --api-secret <your api secret> --schema <your-schema/channel ID> --ssid <your WiFi name> --password <your WiFi password> --region eu

Je n'ai pas de message d'erreur, je reçois bien l'id, la clé et l'adresse IP de mon domicile. 

Désolé pour mon anglais, je suis français.

**Comment by codetheweb** [comment]
Are you saying that the command line tool outputs the device ID/key, but it doesn't appear on the Tuya dashboard?  Why does it need to? Does TuyAPI work with it correctly?

If I'm misunderstanding and `tuya-cli` just doesn't output anything, I'm planning to implement #323 this weekend which may help.

**Comment by codetheweb** [comment]
#323 has [now been implemented](https://github.com/codetheweb/tuyapi/issues/308#issuecomment-650643347).

Try it out and let me know if that helps.

**Comment by hyperion007** [comment]
I am having issues with #323 and I scan the QR code from the app but all I get is a message in "message center" in the app, that says: "an Unknown Device mobile phone has just logged in to this account..." and the QR code expires as soon as it is scanned but there are no link devices listed on iot.tuya.com

I tried to do it with the other method, tuya-cli link but it cannot handle the "!" that I have in my wifi password... 

Any help on getting this up and running would be much appreciated! Thanks.

**Comment by codetheweb** [comment]
@hyperion007 can you please open a new issue? Wrapping the password in single quotes should work fine.

**Comment by BoneheadFraggle** [comment]
I bought 4 plugs today and I have spent the last 6 hours or so with the exact same QR code expires error as @hyperion007 . Thanks for letting me know I'm not the only one :-)

Regarding the alternative method, it worked for me once (the first plug, no problems at all), but I have tried at least 15 times on the next plug at no avail. Same network, identical plugs, same firmware. 

Can it be problems at Tuya side?
